### ISSUE TITLE
$onAction after() not executed everytime

### ISSUE THREAD

**Issue by LadIQe** [posva/pinia#807]
### Steps to reproduce the behavior

1. Add to store $onAction with callback
2. In callback, use after() function with console.log (for debugging)
3. Use store in some component with detached: true
4. Execute action in store to trigger after() function, if it doesnt work, change detached to false and try again.
5. Try to reload page if after() was successful, then it should not work at all, only after detached is changed to opposite value

### Expected behavior

After() should be called everytime with detached: true or detached: false if component is root (for example App.vue)

### Actual behavior

It works one-time only, then it doesnt work. After detached is changed to opposite value, it works, again one-time only

**Comment by posva** [comment]
Can you create a repro using https://github.com/piniajs/example-vue-3-vite

**Comment by posva** [comment]
Closing due to inactivity. Please open a new issue with a reference to this one if you can follow up with more information.

**Comment by LadIQe** [comment]
yes, sorry for not responding. I got married :) so there was no time for this.

Should I open new issue or am I allowed to continue?

**Comment by posva** [comment]
Congrats! 🎉 
It doesn't matter if you open a new one or continue here, but I do need the reproduction to check!

**Comment by LadIQe** [comment]
Yes I understand. Problem is, that with Vite it looks like its working properly.

With quasar v2 it is little complicated. 2 days ago, it didnt worked at all. Yesterday, it worked without changing code. Today, it doesnt work again, but when I deployed it on develop server, its working.

So I assume, its something on my machine. I will try to create quasar v2 repo with pinia to test it and upload it for you. But for now, its random on my machine

**Comment by posva** [comment]
The repro must be without quasar so I can check it 🙏  Otherwise the problem could come from quasar

**Comment by LadIQe** [comment]
Ok it seems its only with quasar project. I cant reproduce it with vite, vue cli

**Comment by nathanthibert** [comment]
Not sure if this relates to this ticket, but I have had a similar experience with `$onAction`  and `addStoreToDevtools`. On first mount (aka page load), the `$onAction` applies everything correctly. Though if I call the action again, when `triggerSubscription` gets called in `wrapAction`, the last action subscription from `actionSubscriptions` seems to override the `afterCallback`. 

<img width="521" alt="Screen Shot 2021-12-19 at 4 14 46 PM" src="https://user-images.githubusercontent.com/86737316/146691306-9c3352b9-9f39-4687-a18a-35c87b6cd61c.png">
<img width="411" alt="Screen Shot 2021-12-19 at 4 15 07 PM" src="https://user-images.githubusercontent.com/86737316/146691324-5306c898-16b3-4e24-8f71-3dc1799b78b6.png">
<img width="460" alt="Screen Shot 2021-12-19 at 4 15 17 PM" src="https://user-images.githubusercontent.com/86737316/146691328-04e05e06-031b-4904-94b7-be0c8a70e2a3.png">
This is the after callback function overwriting the one I want to apply from the `UserTable.vue`
<img width="478" alt="Screen Shot 2021-12-19 at 4 15 53 PM" src="https://user-images.githubusercontent.com/86737316/146691329-05600a0d-671b-4d9c-9c70-82698b612f81.png">

@posva What do you think might be happening?

**Comment by posva** [comment]
See https://github.com/vuejs/pinia/pull/893

**Comment by nathanthibert** [comment]
> See #893

Awesome. I saw the code was in the v2 branch. (as it didn't match what I had locally). @posva when do you think this change  will hit primetime?
